lib/nx_gzlib.c

```diff
@@ -244,10 +244,10 @@ int nx_gzwrite (gzFile file, const void *buf, unsigned len)
 
 int gzclose(gzFile file)
 {
-	uint8_t sel = is_deflate ?
-		      nx_config.mode.deflate : nx_config.mode.inflate;
+	gz_statep state = (gz_statep) file;
+	z_streamp stream = &(state->strm);
 
-	if (sel == GZIP_SW)
+	if (0 == has_nx_state(stream))
```

Comment:
In PR #169 , I'm proposing to rewrite these gz* functions and reuse the code from deflate*/inflate* family of functions.
I believe one of the patches there helps to simplify this part here.
If there is an agreement on what I proposed there, I suspect this commit could be removed.

Reply:
Ok, I'm keeping the commit for now.